tests/run-separate-compilation/f-interpolation-1/FQuote_1.scala

```diff
@@ -0,0 +1,58 @@
+import scala.quoted._
+import scala.tasty.Reflection
+
+import scala.language.implicitConversions
+
+object FQuote {
+
+  implicit class SCOps(ctx: StringContext) {
+    inline def ff(args: => Any*): String = ~impl('(this), '(args))
+  }
+
+  /*private*/ def impl(receiver: Expr[SCOps], args: Expr[Seq[Any]])(implicit reflect: Reflection): Expr[String] = {
+    import reflect._
+
+    def liftListOfAny(lst: List[Term]): Expr[List[Any]] = lst match {
+      case x :: xs  =>
+        val head = x.seal[Any]
+        val tail = liftListOfAny(xs)
+        '{ ~head :: ~tail }
+      case Nil => '(Nil)
+    }
+
+    def isStringConstant(tree: Term) = tree match {
+      case Term.Literal(_) => true
+      case _ => false
+    }
+
+    def isSCOpsConversion(tree: Term) =
+      tree.symbol.fullName == "FQuote$.SCOps"
+
+    def isStringContextApply(tree: Term) =
+      tree.symbol.fullName == "scala.StringContext$.apply"
+
+    // FQuote.SCOps(StringContext.apply([p0, ...]: String*)
+    val parts = receiver.unseal.underlyingArgument match {
```

Comment:
For later: it seems the name `underlyingArgument` does not make sense here, as it's a receiver, not argument.